rcl/include/rcl/publisher.h

```diff
@@ -397,6 +397,33 @@ RCL_PUBLIC
 bool
 rcl_publisher_is_valid(const rcl_publisher_t * publisher);
 
+/// Get the number of subscriptions matched to a publisher.
+/**
+ * Used to get the internal count of subscriptions matched to a publisher.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | Yes
+ * Uses Atomics       | Maybe [1]
+ * Lock-Free          | Maybe [1]
+ * <i>[1] only if the underlying rmw doesn't make use of this feature </i>
+ *
+ * \param[in] publisher pointer to the rcl publisher
+ * \param[out] subscription_count number of matched subscriptions
+ * \return `RCL_RET_OK` if the count was retrieved, or
+ * \return `RCL_RET_INVALID_ARGUMENT` if any arguments are invalid, or
+ * \return `RCL_RET_SUBSCRIPTION_INVALID` if the subscription is invalid, or
+ * \return `RCL_RET_ERROR` if an unspecified error occurs.
+ */
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rmw_ret_t
```

Comment:
rcl functions should return `rcl_ret_t` rather than `rmw_ret_t`, @mjcarroll can you fix this up?